### PR TITLE
Modify deleteFromDocument description

### LINKED ISSUE #34943: deleteFromDocument deletes more than just text
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Selection/deleteFromDocument

### What specific section or headline is this issue about?

First paragraph

### What information was incorrect, unhelpful, or incomplete?

It says the method deletes the  text in the selection   This implies (to me anyway) that it deleted the text nodes or perhaps empties them out.   In my tests it actually deletes everything in the selection.    I have a div full of text, embedded divs., images etc.  If I select the whole thing and call  deleteFromDocument(), the embedded divs imgs etc are completely removed.    (That's good - that's what was hoping).  

### What did you expect to see?

Clarity on what is actually removed.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/selection/deletefromdocument`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Selection/deleteFromDocument
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/selection/deletefromdocument/index.md
* Last commit: https://github.com/mdn/content/commit/f2f9346c0c0e9f6676f2df9f1850933e274401de
* Document last modified: 2023-04-08T06:00:31.000Z

</details>

### PR DESCRIPTION
**Pull request by ADTmux**
### Description

Modified `en-US/docs/Web/API/Selection/deleteFromDocument` page where it now talks about nodes getting deleted rather than just text. I added a heading in the example because I thought we might want to show more elements than just one paragraph. 

### Motivation

Conveys what the function does better

### Related issues and pull requests

Fixes https://github.com/mdn/content/issues/34943

### CODE REVIEW

**Comment by ADTmux** [comment]
Please let me know any opinions on this. Happy to discuss and learn :)

File: files/en-us/web/api/selection/deletefromdocument/index.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ browser-compat: api.Selection.deleteFromDocument
 {{ ApiRef("DOM") }}
 
 The **`deleteFromDocument()`** method of the
-{{domxref("Selection")}} interface deletes the selected text from the document's DOM.
+{{domxref("Selection")}} interface deletes the selected {{ domxref("Node", "Nodes") }} from the document's DOM.
```
Comment (Josh-Cena) [inline files/en-us/web/api/selection/deletefromdocument/index.md:12]:
```suggestion
{{domxref("Selection")}} interface deletes the selected {{domxref("Node")}} objects from the document's DOM.
```

`Nodes` is not a code entity. `domxref` is going to code-format it. Let's just reword it instead.

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Neither the content before and after is correct. It needs to account for both "partially selected nodes" and "fully selected nodes":

- If a node is only partially selected (can only happen if the node overlaps with the start or end of selection), then only the selected text gets deleted, and the node is kept.
- If a node is fully selected, then the node itself is removed.

I would also suggest you add the same verbiage to https://developer.mozilla.org/en-US/docs/Web/API/Range/deleteContents, and mention on the `deleteFromDocument` that it actually just calls `deleteContents` on the underlying `Range`.

**Comment by Josh-Cena** [comment]
@ADTmux FYI, you still have a review suggestion outstanding.

**Comment by ADTmux** [comment]
Hi @Josh-Cena , I was trying to read up about what you mentioned from [](https://www.w3.org/TR/2000/REC-DOM-Level-2-Traversal-Range-20001113/ranges.html#td-ancestor-container) , part 2.6. I did not add details to deleteFromDocument() since it is only a wrapper.

File: files/en-us/web/api/selection/deletefromdocument/index.md [hunk 2]
```diff
@@ -27,18 +27,19 @@ None ({{jsxref("undefined")}}).
 
 ## Examples
 
-This example lets you delete selected text by clicking a button. Upon clicking the
+This example lets you delete selected nodes by clicking a button. Upon clicking the
```
Comment (Josh-Cena) [inline files/en-us/web/api/selection/deletefromdocument/index.md:30]:
```suggestion
This example lets you delete selected text and/or nodes by clicking a button. Upon clicking the
```

File: files/en-us/web/api/selection/deletefromdocument/index.md [hunk 3]
```diff
@@ -27,18 +27,19 @@ None ({{jsxref("undefined")}}).
...
 button, the
-{{domxref("Window.getSelection()")}} method gets the selected text, and the
+{{domxref("Window.getSelection()")}} method gets the selected objects, and the
```
Comment (Josh-Cena) [inline files/en-us/web/api/selection/deletefromdocument/index.md:32]:
```suggestion
{{domxref("Window.getSelection()")}} method gets the selected content, and the
```

**Review by Josh-Cena — APPROVED** [review]
Thanks! I think this looks good. Great work.

File: files/en-us/web/api/selection/deletefromdocument/index.md [hunk 4]
```diff
@@ -27,18 +27,19 @@ None ({{jsxref("undefined")}}).
...
 ### HTML
 
 ```html
+<p>Try highlighting some of the elements in these paragraphs.</p>
 <p>
-  Try highlighting some of the text in this paragraph. Once you do, you can
-  remove the selected content by clicking the button below.
+  Once you do, you can remove the selected content by clicking the button below.
 </p>
+<h2>All of the selected nodes are removed.</h2>
```
Comment (Josh-Cena) [inline files/en-us/web/api/selection/deletefromdocument/index.md:42]:
```suggestion
<h2>All of the selected text/nodes are removed.</h2>
```

File: files/en-us/web/api/selection/deletefromdocument/index.md [hunk 5]
```diff
@@ -27,18 +27,19 @@ None ({{jsxref("undefined")}}).
...
 ### HTML
 
 ```html
+<p>Try highlighting some of the elements in these paragraphs.</p>
 <p>
-  Try highlighting some of the text in this paragraph. Once you do, you can
-  remove the selected content by clicking the button below.
+  Once you do, you can remove the selected content by clicking the button below.
 </p>
+<h2>All of the selected text/nodes are removed.</h2>
```
Comment (Josh-Cena) [inline files/en-us/web/api/selection/deletefromdocument/index.md:42]:
```suggestion
<h2>All of the selected text/nodes will be removed.</h2>
```

**Pull request merged**